### PR TITLE
Update JCB request capability docs link and capitalize it on the transaction details page

### PR DESCRIPTION
**Pull request by mordeth**
#### Changes proposed in this Pull Request
This PR fixes the following issues:
- Updates the JCB capability request docs link button URL.
- Fixes the issue where JCB is not capitalized on the transaction details page.

Before: 
Screenshot 2023-10-06 at 12 36 03

After:
Screenshot 2023-10-06 at 12 34 30

#### Testing instructions
- Connect a Japanese account and create a test order with a JCB [test card](https://stripe.com/docs/testing#asia-pacific) or use my existing test account `acct_1Ny71xQuMKrISNuR`.
- Navigate to `Payments` -> `Transactions`.
- Select the transaction paid with the JCB test card and navigate to the details page.
- Scroll down the `Payment method` section.
- Verify the `Type` value has JCB capitalized.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7404` or branch name `fix/finish-setup-link` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/finish-setup-link) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 4c2f5fd92d8be1a7216c235a03d6875fc23523af
- Build time: 2023-10-06 11:54:11 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by elazzabi — APPROVED** [review]
Looks good 👍

File: changelog/fix-finish-setup-link [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
```
Comment (anu-rock) [inline changelog/fix-finish-setup-link:2]: This doesn't look like a dev change but a user-facing one.

**Review by anu-rock — APPROVED** [review]
LGTM 👍

File: changelog/fix-finish-setup-link (hunk 1 above)
Comment (mordeth) [reply changelog/fix-finish-setup-link:2]: It is a dev note because the JCB has not been released yet.

Comment (anu-rock) [reply changelog/fix-finish-setup-link:2]: I could be wrong, but AFAIK "dev" changes refer to ones related to developer experience like tooling etc.

**Pull request merged**